plugins/woocommerce/includes/abstracts/abstract-wc-product.php

```diff
@@ -2021,7 +2021,14 @@ public function add_to_cart_description() {
 	public function get_image( $size = 'woocommerce_thumbnail', $attr = array(), $placeholder = true ) {
 		$image = '';
 		if ( $this->get_image_id() ) {
-			$image = wp_get_attachment_image( $this->get_image_id(), $size, false, $attr );
+			$image_alt = get_post_meta( $this->get_image_id(), '_wp_attachment_image_alt', true );
+			$attr      = wp_parse_args(
+				$attr,
+				array(
+					'alt' => $image_alt ? $image_alt : $this->get_name(),
```

Comment:
This could be slightly simplified with the shorthand ternary operator or the null coalescing operator (I'm unsure about the possible return types of `get_post_meta()`).
```php
'alt' => $image_alt ?: $this->get_name(),
```
```php
'alt' => $image_alt ?? $this->get_name(),
```

Also, lines 2027-2029 could be merged into one for better readability. Maybe something like this:
```php
array( 'alt' => $image_alt ?: $this->get_name() ) // or `??`
```